@MartinHjelmare Pylint fails with `homeassistant/components/proxmoxve/binary_sensor.py:22:16: C0103: Variable name "vm" doesn't conform to snake_case naming style (invalid-name)`

Can I just disable this error using `# pylint: disable=invalid-name`?

--- a/.coveragerc
+++ b/.coveragerc
@@ -508,7 +508,7 @@ omit =
     homeassistant/components/proliphix/climate.py
     homeassistant/components/prometheus/*
     homeassistant/components/prowl/notify.py
-    homeassistant/components/proxmox/*
+    homeassistant/components/proxmoxve/*
     homeassistant/components/proxy/camera.py
     homeassistant/components/ptvsd/*
     homeassistant/components/pulseaudio_loopback/switch.py

--- a/homeassistant/components/proxmoxve/__init__.py
+++ b/homeassistant/components/proxmoxve/__init__.py
@@ -17,10 +17,10 @@ from homeassistant.const import (
     CONF_VERIFY_SSL,
 )
 
-PROXMOX_CLIENTS = {}
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = "proxmoxve"
+PROXMOX_CLIENTS = "proxmox_clients"
 CONF_REALM = "realm"
 CONF_NODE = "node"
 CONF_NODES = "nodes"
@@ -90,14 +90,17 @@ def setup(hass, config):
             proxmox_client = ProxmoxClient(
                 host, port, user, realm, password, verify_ssl
             )
+            proxmox_client.build_client()
         except AuthenticationError:
             _LOGGER.warning("Invalid credentials")
+            return False
 
         if proxmox_client is None:
             _LOGGER.warning("Failed to connect to proxmox")
             return False
 
-        PROXMOX_CLIENTS[f"{host}:{str(port)}"] = proxmox_client
+        hass.data[PROXMOX_CLIENTS] = {}
+        hass.data[PROXMOX_CLIENTS][f"{host}:{str(port)}"] = proxmox_client
 
     hass.helpers.discovery.load_platform(
         "binary_sensor", DOMAIN, {"entries": config[DOMAIN]}, config
@@ -126,8 +129,6 @@ class ProxmoxClient:
         self._password = password
         self._verify_ssl = verify_ssl
 
-        self.build_client()
-
     def build_client(self):
         """Construct the ProxmoxAPI client."""
 
@@ -142,7 +143,7 @@ class ProxmoxClient:
         self._connection_start_time = time.time()
 
     def get_api_client(self):
-        """Return the ProxmoxAPI client (and rebuilds it if required)."""
+        """Return the ProxmoxAPI client and rebuild it if necessary."""
 
         connection_age = time.time() - self._connection_start_time
 

--- a/homeassistant/components/proxmoxve/__init__.py
+++ b/homeassistant/components/proxmoxve/__init__.py
@@ -129,6 +129,9 @@ class ProxmoxClient:
         self._password = password
         self._verify_ssl = verify_ssl
 
+        self._proxmox = None
+        self._connection_start_time = None
+
     def build_client(self):
         """Construct the ProxmoxAPI client."""
 

--- a/homeassistant/components/proxmoxve/binary_sensor.py
+++ b/homeassistant/components/proxmoxve/binary_sensor.py
@@ -19,20 +19,20 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         sensors = []
 
         for node in entry[CONF_NODES]:
-            for vm in node[CONF_VMS]:
+            for virtual_machine in node[CONF_VMS]:
                 sensors.append(
                     ProxmoxBinarySensor(
-                        PROXMOX_CLIENTS[f"{entry[CONF_HOST]}:{str(port)}"],
+                        hass.data[PROXMOX_CLIENTS][f"{entry[CONF_HOST]}:{str(port)}"],
                         node["node"],
                         ProxmoxItemType.qemu,
-                        vm,
+                        virtual_machine,
                     )
                 )
 
             for container in node[CONF_CONTAINERS]:
                 sensors.append(
                     ProxmoxBinarySensor(
-                        PROXMOX_CLIENTS[f"{entry[CONF_HOST]}:{str(port)}"],
+                        hass.data[PROXMOX_CLIENTS][f"{entry[CONF_HOST]}:{str(port)}"],
                         node["node"],
                         ProxmoxItemType.lxc,
                         container,
@@ -47,19 +47,13 @@ class ProxmoxBinarySensor(BinarySensorDevice):
 
     def __init__(self, proxmox_client, item_node, item_type, item_id):
         """Initialize the binary sensor."""
-
         self._proxmox_client = proxmox_client
         self._item_node = item_node
         self._item_type = item_type
         self._item_id = item_id
 
-        item = self.poll_item()
-
-        if item is None:
-            _LOGGER.warning("Couldn't find VM/Container with the ID %s", self._item_id)
-
-        self._vmname = item["name"]
-        self._name = f"{self._item_node} {self._vmname} running"
+        self._vmname = None
+        self._name = None
 
         self._state = None
 
@@ -86,7 +80,13 @@ class ProxmoxBinarySensor(BinarySensorDevice):
 
     def update(self):
         """Check if the VM/Container is running."""
-        self._state = self.poll_item()["status"] == "running"
+        item = self.poll_item()
+
+        if item is None:
+            _LOGGER.warning("Failed to poll VM/container %s", self._item_id)
+            return
+
+        self._state = item["status"] == "running"
 
     def poll_item(self):
         """Find the VM/Container with the set item_id."""
@@ -98,5 +98,14 @@ class ProxmoxBinarySensor(BinarySensorDevice):
         item = next(
             (item for item in items if item["vmid"] == str(self._item_id)), None
         )
+
+        if self._vmname is None:
+            self._vmname = item["name"]
+
+        if self._name is None:
+            self._name = f"{self._item_node} {self._vmname} running"
+
+        if item is None:
+            _LOGGER.warning("Couldn't find VM/Container with the ID %s", self._item_id)
 
         return item

